src/NugetFeedDeploymentRepository/NugetFeedDeploymentRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Etg.Yams.NuGet.Storage
+{
+    public class NugetFeedDeploymentRepository : IDeploymentRepository
+    {
+        private SourceRepository _sourceRepository;
+
+        public NugetFeedDeploymentRepository(string feedUrl = "https://api.nuget.org/v3/index.json")
+        {
+            List<Lazy<INuGetResourceProvider>> providers = new List<Lazy<INuGetResourceProvider>>();
+            providers.AddRange(Repository.Provider.GetCoreV3());
+            PackageSource packageSource = new PackageSource(feedUrl);
+            _sourceRepository = new SourceRepository(packageSource, providers);
+        }
+
+        public Task DeleteApplicationBinaries(AppIdentity appIdentity)
+        {
+            throw new NotSupportedException("NuGet feeds are immutable.");
+        }
+
+        public async Task DownloadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            DownloadResource downloadResource = await _sourceRepository.GetResourceAsync<DownloadResource>();
+            PackageIdentity packageIdentity = new PackageIdentity(appIdentity.Id, new NuGetVersion(appIdentity.Version.ToString()));
+            var result = await downloadResource.GetDownloadResourceResultAsync(packageIdentity, new PackageDownloadContext(new SourceCacheContext()), null, new Logger(), CancellationToken.None);
+        }
+
+        public async Task<bool> HasApplicationBinaries(AppIdentity appIdentity)
+        {
+            PackageMetadataResource packageMetadataResource = await _sourceRepository.GetResourceAsync<PackageMetadataResource>();
+            PackageIdentity packageIdentity = new PackageIdentity(appIdentity.Id, new NuGetVersion(appIdentity.Version.ToString()));
+            IPackageSearchMetadata searchMetadata = await packageMetadataResource.GetMetadataAsync(packageIdentity, new SourceCacheContext(), new Logger(), CancellationToken.None);
+
+            return searchMetadata?.IsListed ?? false;
+        }
+
+        public Task UploadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            throw new NotSupportedException("Yams doesn't support publishing NuGet packages. Publish your packages from your build server.");
```

Comment:
> Yams doesn't support publishing NuGet packages

It's more that this repository doesn't support publishing. In fact, why not? Users of `IDeploymentRepository` that are using it to upload binaries will not be able to use this plugin.

Reply:
I don't think the upload feature is needed that much and it will be more complex due to the need for credentials. I was hoping I could implement it later.

Reply:
That's fine. Just to give an example, the `InstallApplicationsCmdlet` calls the upload Api and will not work with this repository if it doesn't support publishing.